### PR TITLE
Fix HID API sending path through pyhidapi

### PR DESCRIPTION
**Pull request by dirkhillbrecht**
Hi,

I found that HID API is not accessed as its API says: The data buffer must contain the report ID as first byte, "0" if there are no distinct report IDs. Furthermore, I only had success sending data to the display if I also chunked them into parts of 64 bytes. The HID API documentation is a bit unclear about this, but I had errors otherwise.

I also added a "-H" option to force the HID API send path. Through USB Core, I could not send more than 128 bytes to the display (Sertronics import from August 2023). This parameter addition might seem a bit clumsy, I am by no means an experienced Python programmer. Actually, I am almost no Python programmer at all and, frankly spoken, I do not like this language at all...

My journey to this solution was: I encountered the problem described in Issue 36, applied that fix. Then I ran in Issue 40 (and - now - also 41). I learned that on any modern Linux (Ubuntu 23.04), pyhidapi is never used as it is either not found or the HID C library lies in another directory as scanned by pyhidapi. I fixed all that so that pyhidapi was actually used, only to find that it also does not work correctly. Finally, I found the discrepancy between pyhidapi's API doc and its usage in the script and fixed that by this patch. Finally, I can write arbitrary long data to the display.

And here we are. I think this patch is correct but generally, it is rather complicated to get this up and running. And I have absolutely no idea why usb.core does not work at all.

Feel free to contact me in case of any questions!

Regards,
Dirk

### CODE REVIEW

File: led-badge-11x44.py [hunk 1]
```diff
@@ -453,6 +454,8 @@ def header(lengths, speeds, modes, blink, ants, brightness=100):
   devinfo = pyhidapi.hid_enumerate(0x0416, 0x5020)
   #dev = pyhidapi.hid_open(0x0416, 0x5020)
 else:
+  if args.hid != "0":
+    sys.exit("HID API access is needed but not initialized. Fix your setup")
```
Comment (jnweiger) [inline led-badge-11x44.py:458]: What are the expected actions to "Fix your setup" ?
I guess some experts would know, but we need Usage messages for inexperience target audience pleases.

Comment (pablitovicente) [reply led-badge-11x44.py:458]: In my case for Ubuntu 22.04 to fix it I had to:

- `sudo apt-get install -y libhidapi-libusb0`
-  ln -s /usr/lib/x86_64-linux-gnu/libhidapi-libusb.so.0 libhidapi-libusb.so.0

**Comment by jnweiger** [comment]
there are conflicts. Did you start on a slightly oder version than current master?

**Review by jnweiger — APPROVED** [review]
looks good, after fixing conflicts.
Thanks!

**Pull request merged**